File: determine-if-string-halves-are-alike/determine-if-string-halves-are-alike.js
/**
 * @param {string} s
 * @return {boolean}
 */
var halvesAreAlike = function(s) {
    let vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];
    
    // Split the word
    let half = s.length/2;
    let partA = s.slice(0,half);
    let partB = s.slice(half, s.lenght);
    
    // Count vowels
    let countA = 0;
    let countB = 0;
    for (let i=0; i<partA.length; i++){
        if (vowels.includes(partA[i])){
            countA ++;
        };
        if (vowels.includes(partB[i])){
            countB ++;
        };
    };
    return countA === countB; 
};

File: determine-if-string-halves-are-alike/determine-if-string-halves-are-alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        vowels= ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
        
        ## split the word
        half = len(s)//2
        part_a = s[:half]
        part_b = s[half:]
        
        ##  count vowels in each part
        count_a = 0
        count_b = 0
        
        for elem in range(len(part_a)):
            if part_a[elem] in vowels:
                count_a += 1
            if part_b[elem] in vowels:
                count_b += 1
        
        return count_a == count_b
            
